## Suggested GitHub folder structure

```text
earthquake_project/
├── data/
│   └── parsed.csv
├── notebooks/
│   └── pandas_earthquake_practice.ipynb
└── README.md
```


In [9]:
import pandas as pd


file_path = '/content/earthquake_project/data/parsed.csv'



df = pd.read_csv(file_path)
df.head()


,alert,cdi,code,detail,dmin,felt,gap,ids,mag,magType,...,status,time,title,tsunami,type,types,tz,updated,url,parsed_place
0,NaN,NaN,37389218,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.008693,NaN,85.0,",ci37389218,",1.35,ml,...,automatic,1539475168010,"M 1.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475395144,https://earthquake.usgs.gov/earthquakes/eventp...,California
1,NaN,NaN,37389202,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.020030,NaN,79.0,",ci37389202,",1.29,ml,...,automatic,1539475129610,"M 1.3 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475253925,https://earthquake.usgs.gov/earthquakes/eventp...,California
2,NaN,4.4,37389194,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.021370,28.0,21.0,",ci37389194,",3.42,ml,...,automatic,1539475062610,"M 3.4 - 8km NE of Aguanga, CA",0,earthquake,",dyfi,focal-mechanism,geoserve,nearby-cities,o...",-480.0,1539536756176,https://earthquake.usgs.gov/earthquakes/eventp...,California
3,NaN,NaN,37389186,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.026180,NaN,39.0,",ci37389186,",0.44,ml,...,automatic,1539474978070,"M 0.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475196167,https://earthquake.usgs.gov/earthquakes/eventp...,California
4,NaN,NaN,73096941,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.077990,NaN,192.0,",nc73096941,",2.16,md,...,automatic,1539474716050,"M 2.2 - 10km NW of Avenal, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,scit...",-480.0,1539477547926,https://earthquake.usgs.gov/earthquakes/eventp...,California


## 2(a) 95th percentile of earthquake magnitude in Japan using `magType == 'mb'`


In [3]:
japan_mb = df[(df['parsed_place'] == 'Japan') & (df['magType'] == 'mb')]
japan_95 = japan_mb['mag'].quantile(0.95)
print('95th percentile:', japan_95)



95th percentile: 4.9


**Answer:** The 95th percentile of `mb` earthquake magnitude in Japan is **4.9**.


## 2(b) Percentage of earthquakes in Indonesia coupled with tsunamis


In [10]:
indonesia = df[df['parsed_place'] == 'Indonesia']
indonesia_tsunami_pct = indonesia['tsunami'].mean() * 100

print('Total Indonesia earthquakes:', len(indonesia))
print('Indonesia earthquakes with tsunami:', indonesia['tsunami'].sum())
print(f'Percentage with tsunami: {indonesia_tsunami_pct:.2f}%')



Total Indonesia earthquakes: 147
Indonesia earthquakes with tsunami: 34
Percentage with tsunami: 23.13%


**Answer:** 34 of 147 Indonesian earthquakes were coupled with tsunamis, or approximately **23.13%**.


## 2(c) Summary statistics for earthquakes in Nevada


In [5]:
nevada = df[df['parsed_place'] == 'Nevada']
nevada.describe()


,cdi,dmin,felt,gap,mag,mmi,nst,rms,sig,time,tsunami,tz,updated
count,15.000000,681.000000,15.000000,681.000000,681.000000,1.00,681.000000,681.000000,681.000000,6.810000e+02,681.0,681.0,6.810000e+02
mean,2.440000,0.166199,2.400000,153.668120,0.500073,2.84,12.618209,0.151986,10.970631,1.538314e+12,0.0,-480.0,1.538402e+12
std,0.501142,0.166228,4.626013,68.735302,0.696710,NaN,9.866963,0.084662,19.607150,5.965637e+08,0.0,0.0,6.010951e+08
min,2.000000,0.001000,1.000000,29.140000,-0.500000,2.84,3.000000,0.000500,0.000000,1.537247e+12,0.0,-480.0,1.537307e+12
25%,2.000000,0.053000,1.000000,97.380000,-0.100000,2.84,6.000000,0.106900,0.000000,1.537854e+12,0.0,-480.0,1.537928e+12
50%,2.200000,0.112000,1.000000,149.140000,0.400000,2.84,10.000000,0.146300,2.000000,1.538280e+12,0.0,-480.0,1.538428e+12
75%,2.900000,0.233000,1.000000,199.720000,0.900000,2.84,16.000000,0.187100,12.000000,1.538821e+12,0.0,-480.0,1.538878e+12
max,3.300000,1.414000,19.000000,355.910000,2.900000,2.84,61.000000,0.863400,129.000000,1.539461e+12,0.0,-480.0,1.539483e+12


There are **681** Nevada earthquake observations. For the `mag` column specifically: mean ≈ **0.5001**, standard deviation ≈ **0.6967**, minimum = **-0.5**, median = **0.4**, and maximum = **2.9**.


## 2(d) Add a Ring of Fire indicator column


In [6]:
ring_of_fire_locations = [
    'Bolivia', 'Chile', 'Ecuador', 'Peru', 'Costa Rica', 'Guatemala',
    'Mexico', 'Japan', 'Philippines', 'Indonesia', 'New Zealand',
    'Canada', 'Fiji', 'Alaska', 'Washington', 'California', 'Russia',
    'Taiwan', 'Tonga', 'Kermadec Islands'
]


df['ring_of_fire'] = (
    df['parsed_place'].isin(ring_of_fire_locations)
    | df['parsed_place'].str.contains('Antarctic', case=False, na=False)
)

df[['parsed_place', 'ring_of_fire']].head()


,parsed_place,ring_of_fire
0,California,True
1,California,True
2,California,True
3,California,True
4,California,True


## 2(e) Number of earthquakes inside and outside Ring of Fire locations


In [7]:
ring_counts = df['ring_of_fire'].value_counts()
print(ring_counts)

print('In Ring of Fire:', df['ring_of_fire'].sum())
print('Outside Ring of Fire:', (~df['ring_of_fire']).sum())



ring_of_fire
True     7188
False    2144
Name: count, dtype: int64
In Ring of Fire: 7188
Outside Ring of Fire: 2144


**Answer:** **7,188** earthquakes occurred in the specified Ring of Fire locations, and **2,144** occurred outside them.


## 2(f) Tsunami count along the Ring of Fire


In [8]:
ring_tsunami_count = df.loc[df['ring_of_fire'], 'tsunami'].sum()
print('Tsunami count along the Ring of Fire:', ring_tsunami_count)



Tsunami count along the Ring of Fire: 45


**Answer:** The tsunami count along the specified Ring of Fire locations is **45**.
